File: python/connect_april.py
"""
AprilAPI 1.0.0

Module to work with april

Author : Merwin
"""

import json
import sys


class April:
    def __init__(self):
        pass

    def recv_request(self):
        n = sys.argv[-1]
        f = open("." + str(current_id) + ".input", "r")
        d = json.loads(f.read())
        f.close()

        return d

    def send_response(self, response):
        n = sys.argv[-1]
        f = "." + str(n) + ".output"
        f = open(f, "w")
        f.write(json.dumps(response))
        f.close()

File: perl/connect_april.pl
use strict;
use warnings;
use JSON;

package ConnectApril;

sub recv_request {
    my $n = $ARGV[$#ARGV];
    my $file_path = ".$n.input";

    eval {
        open my $file, '<', $file_path or die $!;
        local $/;
        my $data = <$file>;
        close $file;
        return JSON::decode_json($data);
    } or do {
        my $err = $@;
        warn $err;
        return;
    };
}

sub send_response {
    my ($response) = @_;
    my $n = $ARGV[$#ARGV];
    my $file_path = ".$n.output";

    eval {
        open my $file, '>', $file_path or die $!;
        print $file JSON::encode_json($response);
        close $file;
    } or do {
        my $err = $@;
        warn $err;
    };
}

1;